Assets/Scripts/UI/DialogBox/PromptOrInfo/DialogBoxPromptOrInfo.cs

```diff
@@ -28,22 +28,27 @@ public class DialogBoxPromptOrInfo : DialogBox
     /// <summary>
     /// Define the buttons that should appear in the dialog (yes, no, cancel).
     /// </summary>
-    /// <param name="buttonsToSet">An combination enum built with bitwise ORs to define the buttons.</param>
-    public void SetButtons(DialogBoxResult buttonsToSet)
+    public void SetButtons(params DialogBoxResult[] buttonsToSet)
     {
-        gameObject.transform.Find("Buttons").gameObject.SetActive(true);
+        Transform buttons = gameObject.transform.Find("Buttons");
+        buttons.gameObject.SetActive(true);
 
-        if ((buttonsToSet & DialogBoxResult.Yes) == DialogBoxResult.Yes)
+        foreach (Transform button in buttons)
+        {
+            button.gameObject.SetActive(false);
+        }
+
+        if (Array.Exists(buttonsToSet, element => element == DialogBoxResult.Yes))
```

Comment:
Since regrettably Unity doesn't serialize dictionaries in the inspector, in order to do this I would have to set up each entry for the dictionary manually in a start method. I could do it if you're dead set on it, but it seems ugly...